### PR TITLE
UCT/IB: Add DV CQE 128bit support

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8218/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8219/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8227/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8229/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_iface.c [hunk 1]
```diff
@@ -634,20 +604,25 @@ struct ibv_cq *uct_ib_create_cq(struct ibv_context *context, int cqe,
...
+    }
+
+    *cq_p = cq;
+    return UCS_OK;
 }
 
-static ucs_status_t uct_ib_iface_create_cq(uct_ib_iface_t *iface, int cq_length,
-                                           size_t *inl, int preferred_cpu,
-                                           int flags, struct ibv_cq **cq_p)
+static ucs_status_t __uct_ib_iface_create_cq(uct_ib_iface_t *iface, int cq_length,
```
Comment (brminich) [inline src/uct/ib/base/ib_iface.c]: minor: I'd remove `__` from the name and change existing `uct_ib_iface_create_cq` to something like `uct_ib_verbs_iface_create_cq`.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8237/ for details (Mellanox internal link).

File: src/uct/ib/base/ib_iface.c [hunk 2]
```diff
@@ -634,20 +604,25 @@ struct ibv_cq *uct_ib_create_cq(struct ibv_context *context, int cqe,
 #else
     cq = ibv_create_cq(context, cqe, NULL, channel, comp_vector);
 #endif
-    return cq;
+    if (!cq) {
+        ucs_error("ibv_create_cq(cqe=%d) failed: %m", cqe);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    *cq_p = cq;
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c]: should this function also update *inl?

File: src/uct/ib/base/ib_iface.c [hunk 3]
```diff
@@ -610,6 +610,7 @@ ucs_status_t uct_ib_verbs_create_cq(struct ibv_context *context, int cqe,
     }
 
     *cq_p = cq;
+    *inl  = 32;
```
Comment (yosefe) [inline src/uct/ib/base/ib_iface.c]: i think it should be 0 for verbs

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8259/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8262/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8322/ for details (Mellanox internal link).

**Pull request merged**